cypress-io/cypress
# fix: process_profiler follow up work for v10

- Closes #22022

### User facing changelog


* Updated the process profiler to output Cypress's summarized CPU and memory metrics when the 'cypress:server:util:process_profiler' debug stream is enabled.

### Additional details


The plugin group had stopped working due to some of the called APIs in the ProjectLifecycleManager being removed. I added a few APIs to enable the profiler to once again access this process info and properly group it. 

### Steps to test


1. Launch cypress with `DEBUG=cypress:server:util:process_profiler`
2. Open a project (`launchpad`), select testing type
3. View process profiler output, ensure 'plugin' group is visible and populated

### How has the user experience changed?


These changes align with our currently documented behavior.

### PR Tasks


- [n/a] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? 
- [n/a] Have API changes been updated in the [`type definitions`](https://github.com/cypress-io/cypress/blob/develop/cli/types/cypress.d.ts)?
